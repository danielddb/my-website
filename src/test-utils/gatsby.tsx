import { FixedObject } from 'gatsby-image';

export const createGatsbyFixedImageMock = (): FixedObject => ({
  base64:
    'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAAMBAAAAAAAAAAAAAAAAAAECBAX/xAAXAQEBAQEAAAAAAAAAAAAAAAABAwAC/9oADAMBAAIQAxAAAAHPVWbVIOrmLbUHn//EABwQAAICAgMAAAAAAAAAAAAAAAABAhEDEBIhMf/aAAgBAQABBQKy6b94tbydmGCHBM//xAAUEQEAAAAAAAAAAAAAAAAAAAAg/9oACAEDAQE/AR//xAAVEQEBAAAAAAAAAAAAAAAAAAABEP/aAAgBAgEBPwEjP//EABoQAAIDAQEAAAAAAAAAAAAAAAARARAxIVL/2gAIAQEABj8CUSj0cZlYK//EABwQAAIDAAMBAAAAAAAAAAAAAAABESExUWFxkf/aAAgBAQABPyHkexBVwy2JbD0c5vD+HUIh8hFdg6m2pgts8P/aAAwDAQACAAMAAAAQrAeA/8QAGBEAAwEBAAAAAAAAAAAAAAAAAAExEBH/2gAIAQMBAT8QqOijz//EABcRAQEBAQAAAAAAAAAAAAAAAAEAETH/2gAIAQIBAT8QOFjdEht//8QAHhABAQACAgIDAAAAAAAAAAAAAREAITFBUWFxgaH/2gAIAQEAAT8QlUAsjTlUNNinPd8mSN7O+H5hwCbq2jEs+VNXA+J9QcoDUX7856WQ8YuQ6yOGf//Z',
  width: 64,
  height: 64,
  src: '/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg',
  srcSet:
    '/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg 1x,\n/static/6f6ccd0263b69cfd693f71c17be8ac7c/eee8e/avatar.jpg 1.5x,\n/static/6f6ccd0263b69cfd693f71c17be8ac7c/6dc0c/avatar.jpg 2x'
});
