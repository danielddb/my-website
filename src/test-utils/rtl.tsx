import { render, RenderOptions } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import light from '../themes/light';

const Wrapper = ({ children }) => (
  <ThemeProvider theme={light}>{children}</ThemeProvider>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
