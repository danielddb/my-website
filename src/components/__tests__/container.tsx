import { axe } from 'jest-axe';
import React from 'react';
import { render } from 'test-utils/rtl';
import Container from '../container';

test('it should render correctly', () => {
  const { container } = render(<Container>Test</Container>);
  expect(container).toMatchSnapshot();
});

test('it should render correctly at "xs" breakpoint', () => {
  const { container } = render(<Container maxWidth="xs">Test</Container>);
  expect(container).toMatchSnapshot();
});

test('it should render correctly at "sm" breakpoint', () => {
  const { container } = render(<Container maxWidth="sm">Test</Container>);
  expect(container).toMatchSnapshot();
});

test('it should render correctly at "md" breakpoint', () => {
  const { container } = render(<Container maxWidth="md">Test</Container>);
  expect(container).toMatchSnapshot();
});

test('it should render correctly at "lg" breakpoint', () => {
  const { container } = render(<Container maxWidth="lg">Test</Container>);
  expect(container).toMatchSnapshot();
});

test('it should render with no accessibility issues', async () => {
  const { container } = render(<Container>Test</Container>);
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});
