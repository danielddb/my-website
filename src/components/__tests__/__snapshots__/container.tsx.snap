// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render correctly 1`] = `
.c0 {
  margin: 0 auto;
  padding: 0 1.5rem;
}

.c0 > :first-child {
  margin-top: 0;
}

@media (min-width:36em) {
  .c0 {
    max-width: 36em;
  }
}

<div>
  <div
    class="c0"
  >
    Test
  </div>
</div>
`;

exports[`it should render correctly at "lg" breakpoint 1`] = `
.c0 {
  margin: 0 auto;
  padding: 0 1.5rem;
}

.c0 > :first-child {
  margin-top: 0;
}

@media (min-width:75em) {
  .c0 {
    max-width: 75em;
  }
}

<div>
  <div
    class="c0"
  >
    Test
  </div>
</div>
`;

exports[`it should render correctly at "md" breakpoint 1`] = `
.c0 {
  margin: 0 auto;
  padding: 0 1.5rem;
}

.c0 > :first-child {
  margin-top: 0;
}

@media (min-width:62em) {
  .c0 {
    max-width: 62em;
  }
}

<div>
  <div
    class="c0"
  >
    Test
  </div>
</div>
`;

exports[`it should render correctly at "sm" breakpoint 1`] = `
.c0 {
  margin: 0 auto;
  padding: 0 1.5rem;
}

.c0 > :first-child {
  margin-top: 0;
}

@media (min-width:48em) {
  .c0 {
    max-width: 48em;
  }
}

<div>
  <div
    class="c0"
  >
    Test
  </div>
</div>
`;

exports[`it should render correctly at "xs" breakpoint 1`] = `
.c0 {
  margin: 0 auto;
  padding: 0 1.5rem;
}

.c0 > :first-child {
  margin-top: 0;
}

@media (min-width:36em) {
  .c0 {
    max-width: 36em;
  }
}

<div>
  <div
    class="c0"
  >
    Test
  </div>
</div>
`;
